Added glossary to the doc (#3626)
scikit-image/scikit-image: doc/source/glossary.rst

Glossary
========

.. currentmodule:: skimage

Work in progress

.. glossary::

    array 
        Numerical array, provided by the :class:`numpy.ndarray` object. In
        ``scikit-image``, images are NumPy arrays, which dimensions
        correspond to spatial dimensions of the image, and color channels for
        color images. See :ref:`numpy`. 

    contour
    iso-valued contour
        Curve along which a 2-D image has a constant value. The interior
        (resp. exterior) of the contour has values greater (resp. smaller)
        than the contour value. 

    contrast
        Differences of intensity or color in an image, which make objects
        distinguishable. Several functions to manipulate the contrast of an
        image are available in :mod:`skimage.exposure`. See :ref:`exposure`.

    float
    float values
        Representation of real numbers, for example as :obj:`np.float32` or
        :obj:`np.float64`. See :ref:`data_types`. Some operations on images

>         need a float datatype (such as multiplicating image values with

Reviewer: ```suggestion
        need a float datatype (such as multiplying image values with
```

        exponential prefactors in :func:`filters.gaussian`), so that 

>         image of integer type are converted to float type internally. Also

Reviewer: ```suggestion
        images of integer type are often converted to float type internally. Also
```

        see :term:`int` values.

    histogram
        For an image, histogram of intensity values, where the range of
        intensity values is divided into bins and the histogram counts how
        many pixel values fall in each bin. See
        :func:`exposure.histogram`.

    int
    int values
        Representation of integer numbers, which can be signed or not, and
        encoded on one, two, four or eight bytes according to the maximum value
        which needs to be represented. In ``scikit-image``, the most common
        integer types are :obj:`np.int64` (for large integer values) and
        :obj:`np.uint8` (for small integer values, typically images of labels
        with less than 255 labels). See :ref:`data_types`. 

>     labels

Reviewer: ```suggestion
    labels
    label image
```
